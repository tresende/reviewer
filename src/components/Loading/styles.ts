import styled, { css, createGlobalStyle } from 'styled-components'

export const GlobalStyle = createGlobalStyle`
  body {
    overflow:hidden
  }
`

export const Wrapper = styled.div`
  ${({ theme }) => css`
    position: fixed;
    width: 100%;
    height: 100%;
    top: 0;
    background-color: rgba(0, 0, 0, 0.9);
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    color: white;
    font-weight: bold;
    z-index: ${theme.layers.alwaysOnTop};
    text-align: center;
  `};
`

export const Logo = styled.img`
  margin-top: 1rem;
  animation: spin 2000ms linear infinite;
  @keyframes spin {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }
`
