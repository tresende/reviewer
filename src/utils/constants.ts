const SCORE_LABEL = ['PÃ©ssimo', 'Ruim', 'Mediano', 'Bom', 'Perfeito']
const MIN_TEXT_LENGTH = 5
const MAX_RATING = 5
const DEFAULT_SCORE = 3
const ANIMATION_DURATION = 1000

export { SCORE_LABEL, MAX_RATING, ANIMATION_DURATION, MIN_TEXT_LENGTH, DEFAULT_SCORE }
