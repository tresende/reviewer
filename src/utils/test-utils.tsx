import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'

import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'
import { LoadingContext, LoadingContextData, LoadingContextDefaultValues } from 'hooks/use-loading'

type CustomRenderProps = {
  loadingProviderProps?: LoadingContextData
} & Omit<RenderOptions, 'queries'>

const customRender = (
  ui: ReactElement,
  { loadingProviderProps = LoadingContextDefaultValues, ...renderOptions }: CustomRenderProps = {}
) =>
  render(
    <ThemeProvider theme={theme}>
      <LoadingContext.Provider value={loadingProviderProps}>{ui}</LoadingContext.Provider>
    </ThemeProvider>,
    renderOptions
  )

export * from '@testing-library/react'
export { customRender as render }
