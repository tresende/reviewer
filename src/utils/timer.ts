export function timeout(ms?: number | undefined) {
  return new Promise((resolve) => setTimeout(resolve, ms))
}
