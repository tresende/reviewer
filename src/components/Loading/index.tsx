import { useLoading } from 'hooks/use-loading'
import * as S from './styles'

const Loading = () => {
  const { loading, text } = useLoading()
  if (!loading) return null
  return (
    <S.Wrapper>
      {text}
      <S.Logo src="/img/borc-logo-transparente.png" />
    </S.Wrapper>
  )
}

export default Loading
