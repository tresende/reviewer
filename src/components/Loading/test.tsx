import { render, screen } from 'utils/test-utils'
import Loading from '.'

describe('<Loading />', () => {
  it('should render empty component when loading context is false', () => {
    const { container } = render(<Loading />)
    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render empty component when loading context is false', () => {
    const loadingProviderProps = {
      start: jest.fn(),
      stop: jest.fn(),
      loading: true,
      text: 'Carregando...'
    }

    render(<Loading />, { loadingProviderProps })
    screen.getByText(/Carregando.../i).click()
  })
})
