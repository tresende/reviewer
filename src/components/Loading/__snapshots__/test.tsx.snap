// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render empty component when loading context is false 1`] = `null`;
